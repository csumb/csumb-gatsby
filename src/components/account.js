import React from 'react'
import styled from 'react-emotion'
import Link from 'gatsby-link'
import theme from 'components/styles/theme'
import PropTypes from 'prop-types'

const AccountSidebarOptions = styled('ul')`
  list-style-type: none;
  margin: 0;
`

const AccountSidebarLink = styled(Link)`
  display: block;
  padding: 0.5rem;
  border-bottom: 1px solid ${theme.colors.black};
  text-decoration: none;
  color: ${theme.colors.primary.dark};
  &[aria-current='page'] {
    background: ${theme.colors.primary.dark};
    color: ${theme.colors.white};
  }
`

const AccountTitle = styled('h2')`
  margin-top: 0;
`

const AccountData = styled('p')`
  font-size: 1.4rem;
  font-weight: bold;
  margin-left: 1rem;
`

const AccountSidebar = props => (
  <AccountSidebarOptions>
    <AccountSidebarLink to="/account">Your account</AccountSidebarLink>
    {props.user._isEmployee && (
      <AccountSidebarLink to="/account/profile">
        Your public profile
      </AccountSidebarLink>
    )}
    {props.user._isApplicant && (
      <AccountSidebarLink to="/account/applicant-status">
        Your application status
      </AccountSidebarLink>
    )}
    <AccountSidebarLink to="/account/messages">Messages</AccountSidebarLink>
    <AccountSidebarLink to="/account/card">Otter Card</AccountSidebarLink>
    <AccountSidebarLink to="/account/print">Printer paper</AccountSidebarLink>
  </AccountSidebarOptions>
)

const AccountGroupWrapper = styled('div')`
  border: 1px solid ${theme.colors.gray.light};
  padding: 1rem;
  margin-bottom: 1rem;
`

const AccountGroupLegend = styled('h3')`
  font-size: 1.5rem;
`

const AccountGroup = props => (
  <AccountGroupWrapper>
    <AccountGroupLegend>{props.legend}</AccountGroupLegend>
    {props.children}
  </AccountGroupWrapper>
)

AccountGroup.propTypes = {
  legend: PropTypes.string.isRequired,
}

export { AccountSidebar, AccountTitle, AccountData, AccountGroup }
