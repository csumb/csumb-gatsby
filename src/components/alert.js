import React from 'react'
import styled from '@emotion/styled'
import Alert from '@reach/alert'
import { colors } from 'style/theme'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import color from 'color'
import VisuallyHidden from 'components/visually-hidden'
import {
  faCheckCircle,
  faExclamationTriangle,
  faExclamationCircle,
  faTimesCircle,
} from '@fortawesome/free-solid-svg-icons'
import { Flex, Box } from '@rebass/grid/emotion'
import bp from './styles/breakpoints'

const baseStyle = `
  display: block;
  padding: 0.5rem;
  margin: 1rem 0;
  p {
    margin-bottom: 0;
  }
`

const AlertIcon = styled(FontAwesomeIcon)`
  font-size: 2.8rem;
  margin-top: 0.5rem;
`

const AlertIconBox = styled(Box)`
  ${bp({
    display: ['none', 'block', 'block', 'block'],
  })};
`

const AlertContent = ({ icon, children }) => (
  <Flex flexWrap="wrap">
    <AlertIconBox width={[0, 1 / 6, 1 / 10]} px={2}>
      <AlertIcon icon={icon} />
    </AlertIconBox>
    <Box width={[1, 5 / 6, 9 / 10]} px={4}>
      {children}
    </Box>
  </Flex>
)

const AlertDangerElement = styled(Alert)`
  ${baseStyle};
  background: ${color(colors.indicators.high)
    .lighten(0.8)
    .hex()};
  border-left: 8px solid ${colors.indicators.high};
`

const AlertDanger = props => (
  <AlertDangerElement {...props}>
    <VisuallyHidden>Danger</VisuallyHidden>
    <AlertContent icon={faExclamationCircle}>{props.children}</AlertContent>
  </AlertDangerElement>
)

const AlertWarningElement = styled(Alert)`
  ${baseStyle};
  background: ${color(colors.indicators.medium)
    .lighten(0.95)
    .hex()};
  border-left: 8px solid ${colors.indicators.medium};
`
const AlertWarning = props => (
  <AlertWarningElement {...props}>
    <VisuallyHidden>Warning</VisuallyHidden>
    <AlertContent icon={faExclamationTriangle}>{props.children}</AlertContent>
  </AlertWarningElement>
)

const AlertSuccessElement = styled(Alert)`
  ${baseStyle};
  background: ${color(colors.indicators.low)
    .lighten(1.5)
    .hex()};

  border-left: 8px solid ${colors.indicators.low};
`

const AlertSuccess = props => (
  <AlertSuccessElement {...props}>
    <VisuallyHidden>For your information</VisuallyHidden>
    <AlertContent icon={faCheckCircle}>{props.children}</AlertContent>
  </AlertSuccessElement>
)

const AlertEmptyElement = styled(Alert)`
  ${baseStyle};
  border: 1px solid ${colors.muted.bright};
`

const AlertEmpty = props => (
  <AlertEmptyElement {...props}>
    <VisuallyHidden>For your information</VisuallyHidden>
    <AlertContent icon={faTimesCircle}>{props.children}</AlertContent>
  </AlertEmptyElement>
)

const AlertFyiElement = styled(Alert)`
  ${baseStyle};
  border: 1px solid ${colors.muted.bright};
  font-weight: bold;
`

const AlertFyi = props => (
  <AlertFyiElement {...props}>{props.children}</AlertFyiElement>
)

export { AlertDanger, AlertSuccess, AlertWarning, AlertEmpty, AlertFyi }
