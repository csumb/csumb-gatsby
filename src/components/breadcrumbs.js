import React from 'react'
import { css } from 'emotion'
import Link from 'gatsby-link'
import styled from 'react-emotion'

const BreadcrumbItem = styled('li')`
  display: inline-block;
  margin-right: 1rem;
  font-size: 80%;
  a {
    color: ${props => props.theme.colors.blue.dark};
  }
`

const Breadcrumbs = (props) => {
  if(!props.breadcrumbs.length) {
    return null
  }
  const breadcrumbs = JSON.parse(props.breadcrumbs)
  return (
    <ul className={css`
      margin: 0;
      padding: 0;
    `}>
      {breadcrumbs.map(crumb => (
        <BreadcrumbItem key={crumb.href}>
          <Link to={crumb.href}>{crumb.title}</Link>
        </BreadcrumbItem>
      ))}
    </ul>
  )
}

export default Breadcrumbs