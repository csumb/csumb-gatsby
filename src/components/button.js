import LinkInspect from './link-inspect'
import styled from '@emotion/styled'
import { colors } from './styles/theme'

const ButtonStyle = props => {
  if (typeof props.buttonType === 'undefined') {
    props.buttonType = 'default'
  }
  return `
  padding: 0.75rem;
  text-decoration: none;
  cursor: pointer;
  border: none;
  color: ${colors.white} !important;
  background: ${colors.buttons.default};
  &:hover {
    color: ${colors.white};
  }
  ${props.huge &&
    `
    font-size: 4rem;
    padding: 1rem;
  `}

  ${
    props.block
      ? `
    display: block;
    width: 100%;
  `
      : `
    display: inline-block;
  `
    }
    ${props.extraMargin &&
    `
      margin: 1rem 0;
    `}
  ${
    props.buttonType === 'default'
      ? `
      color: ${colors.white};
      background: ${colors.buttons.default};
      &:hover {
        
      }
    `
      : ``
    } 
  ${
    props.buttonType === 'primary'
      ? `
      color: ${colors.white};
      background: ${colors.buttons.default};
      &:hover {
        
      }
    `
      : ``
    } 
  ${
    props.buttonType === 'highImpact'
      ? `
      color: ${colors.indicators.high};
      border: 3px solid ${colors.indicators.high};
      &:hover {
        background: ${colors.indicators.high};
      }
    `
      : ``
    };
`
}

const ButtonLink = styled(LinkInspect)`
  ${props => ButtonStyle(props)};
`

const Button = styled('button')`
  ${props => ButtonStyle(props)};
`

const LinkyButton = styled('button')`
  background: transparent;
  border: none;
  color: ${colors.primary.dark};
  text-decoration: underline;
  cursor: pointer;
  padding: 0;
  ${props => props.small && `font-size: 80%;`}
`

export { Button, ButtonLink, LinkyButton }
