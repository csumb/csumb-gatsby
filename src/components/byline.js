import styled from '@emotion/styled'

const Byline = styled('div')`
  font-weight: bold;
  display: inline-block;
  margin-bottom: 1rem;
`

export default Byline
