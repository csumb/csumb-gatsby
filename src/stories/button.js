import React from 'react'
import { storiesOf } from '@storybook/react'
import { Button, ButtonLink } from 'components/button'
import typography from 'utils/typography'

typography.injectStyles()

storiesOf('ButtonLink', module)
  .add('Default', () => <ButtonLink to="#here">A button</ButtonLink>, {
    info: `Button links are used for making a link to a page that evokes the feeling
    of an action. Do not use button links to invoke Javascript or do something
    that changes the state of the current page, use a regular button insted.`,
  })
  .add('Primary', () => (
    <ButtonLink to="#here" buttonType="primary">
      A primary button
    </ButtonLink>
  ))
  .add('High impact', () => (
    <ButtonLink to="#here" buttonType="highImpact">
      High impact
    </ButtonLink>
  ))

storiesOf('Button', module)
  .add('Default', () => <Button>A native button</Button>, {
    info: `While styled the same as Button Links, buttons are for in-page actions,
    like popping open a modal or changing a value.`,
  })
  .add('Primary', () => <Button buttonType="primary">A primary button</Button>)
  .add('High Impact', () => (
    <Button buttonType="highImpact">High impact</Button>
  ))
