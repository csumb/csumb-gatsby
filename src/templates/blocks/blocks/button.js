import React from 'react'
import { ButtonLinkAnchor } from '../../../components/common/button'
import styled from '@emotion/styled'

const Button = styled(ButtonLinkAnchor)`
  margin-bottom: 1rem;
  margin-right: 1rem;
`

const BlockButton = ({ url, text }) => (
  <Button
    href={url
      .replace(`https://csumb.edu`, '')
      .replace(`http://csumb.edu`, '')
      .replace('https://edit.csumb.edu', '')}
    buttonType="default"
    className="button"
  >
    {text}
  </Button>
)

export default BlockButton
