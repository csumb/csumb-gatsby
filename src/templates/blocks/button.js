import React from 'react'
import Button from '../../components/button'


class BlockButton extends React.Component {
  render() {
    return (
      <Button href={this.props.block.data.url}>
        {this.props.block.data.text}
      </Button>
    )
  }
}

export default BlockButton