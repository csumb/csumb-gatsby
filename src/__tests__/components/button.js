import React from 'react'
import renderer from 'react-test-renderer'
import { Button, ButtonLink } from 'components/button'

describe('Button', () => {
  it('renders native button correctly', () => {
    const tree = renderer.create(<Button>A button!</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders default button type correctly', () => {
    const tree = renderer
      .create(<Button buttonType="default">A button!</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders default button type correctly', () => {
    const tree = renderer
      .create(<Button buttonType="primary">A button!</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders default button type correctly', () => {
    const tree = renderer
      .create(<Button buttonType="highImpact">A button!</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders huge button type correctly', () => {
    const tree = renderer
      .create(
        <Button buttonType="highImpact" huge>
          A huge button!
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders block button type correctly', () => {
    const tree = renderer
      .create(
        <Button buttonType="highImpact" block>
          A block button!
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('ButtonLink', () => {
  it('renders link button correctly', () => {
    const tree = renderer
      .create(<ButtonLink to="/home">A link button!</ButtonLink>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders nothing if there is no "to" prop', () => {
    const tree = renderer.create(<ButtonLink>A link</ButtonLink>).toJSON()
    expect(tree).toBeNull()
  })
})
