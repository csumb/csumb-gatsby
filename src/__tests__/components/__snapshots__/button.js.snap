// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders block button type correctly 1`] = `
<button
  className="css-1pjuk3c"
>
  A block button!
</button>
`;

exports[`Button renders default button type correctly 1`] = `
<button
  className="css-xxwvww"
>
  A button!
</button>
`;

exports[`Button renders default button type correctly 2`] = `
<button
  className="css-548gsz"
>
  A button!
</button>
`;

exports[`Button renders default button type correctly 3`] = `
<button
  className="css-18gpm1f"
>
  A button!
</button>
`;

exports[`Button renders huge button type correctly 1`] = `
<button
  className="css-9jomw4"
>
  A huge button!
</button>
`;

exports[`Button renders native button correctly 1`] = `
<button
  className="css-xxwvww"
>
  A button!
</button>
`;

exports[`ButtonLink renders link button correctly 1`] = `
<a
  className="css-xxwvww"
  href="/home"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  A link button!
</a>
`;
